src/mlio/csv_record_tokenizer.cxx

```diff
@@ -117,6 +117,29 @@ csv_record_tokenizer::next()
     return true;
 }
 
+inline void
+csv_record_tokenizer::reset()
+{
+    value_.clear();
+    is_truncated_ = false;
+}
+
+inline void
+csv_record_tokenizer::push_char(char chr)
+{
+    if (max_field_length_) {
```

Comment:
As Rahul suggested, this can be simplified to:
```
if (max_field_length_ && value_.size() == *max_field_length_) {
  is_truncated_ = true;
}
else {
  value_.push_back(chr);
}
```